client.go

```diff
@@ -425,7 +429,14 @@ func (cli *Client) SetWSDialer(dialer *websocket.Dialer) {
 // Connect connects the client to the WhatsApp web websocket. After connection, it will either
 // authenticate if there's data in the device store, or emit a QREvent to set up a new link.
 func (cli *Client) Connect() error {
-	err := cli.connect()
+	if cli == nil {
+		return ErrClientIsNil
+	}
+
+	cli.socketLock.Lock()
+	defer cli.socketLock.Unlock()
+
+	err := cli.unlockedConnect()
```

Comment:
What's this refactor for? Does it make any difference?

Reply:
Shoot, that's a leftover from a previous take.